UIforETW/UIforETWDlg.cpp

```diff
@@ -40,10 +41,821 @@ const int kRecordTraceHotKey = 1234;
+				   "Cannot add a non-CWnd-derived class as a tooltip!" );
+
+	const BOOL addToolTipResult = toolTip->AddTool(btToAdd, toolTipMessageToAdd);
+	
+	if (addToolTipResult == TRUE)
+		return;
+	OutputDebugStringA("UIforETW: Failed to add a message to a tooltip!!\r\n");
+	OutputDebugStringA("The tooltip message:\r\n\t`");
+	OutputDebugStringW(toolTipMessageToAdd);
+	OutputDebugStringA("`\r\n");
+	std::terminate();
+}
+
+
+CRect GetWindowRectFromHwnd( _In_ const HWND hwnd )
+{
+	RECT windowRect_temp;
+	ASSERT(::IsWindow(hwnd));
+	const BOOL getWindowRectResult = ::GetWindowRect(hwnd, &windowRect_temp);
+	if (getWindowRectResult == 0)
+	{
+		outputPrintf(L"GetWindowRect failed!!\n");
+		ErrorHandling::outputErrorDebug();
+		std::terminate();
+	}
+	return windowRect_temp;
+}
+
+void SetSaveTraceBuffersWindowText( _In_ const HWND hWnd )
+{
+	const BOOL btSaveTbSetTextResult = ::SetWindowTextW(hWnd, L"Sa&ve Trace Buffers");
+	if (btSaveTbSetTextResult == 0)
+	{
+		outputPrintf(L"::SetWindowTextW( btSaveTraceBuffers_.m_hWnd, "
+					 L"L\"Sa&ve Trace Buffers\" failed!!!\n" );
+		ErrorHandling::outputErrorDebug();
+		std::terminate();
+	}
+}
+
+bool appendAboutBoxToSystemMenu( _In_ const CWnd& window )
+{
+	// Add "About..." menu item to system menu.
+
+	// IDM_ABOUTBOX must be in the system command range.
+	static_assert( ( IDM_ABOUTBOX & 0xFFF0 ) == IDM_ABOUTBOX,
+				   "IDM_ABOUTBOX IS NOT in the system command range!" );
+	static_assert( IDM_ABOUTBOX < 0xF000,
+				   "IDM_ABOUTBOX IS NOT in the system command range!" );
+
+
+
+	//The pointer returned by [CWnd::GetSystemMenu(FALSE)] may be temporary and should not be stored for later use.
+	CMenu* const pSysMenu = window.GetSystemMenu(FALSE);
+	if (pSysMenu == NULL)
+		return false;
+
+	CString strAboutMenu;
+
+	//[CStringT:LoadString returns] nonzero if resource load was successful; otherwise 0.
+	const BOOL bNameValid = strAboutMenu.LoadString(IDS_ABOUTBOX);
+	ATLASSERT(bNameValid);
+	if (bNameValid == 0)
+	{
+		outputPrintf(L"Failed to load about box menu string!!\n");
+		return false;
+	}
+	if (strAboutMenu.IsEmpty())
+	{
+		outputPrintf(L"About box string is empty?!?!\n");
+		return false;
+	}
+
+	//[CWnd::AppendMenu returns] nonzero if the function is successful; otherwise 0.
+	const BOOL appendSeparatorResult = pSysMenu->AppendMenu(MF_SEPARATOR);
+	if (appendSeparatorResult == 0)
+	{
+		outputPrintf(L"Failed to append separator to menu!\n");
+		return false;
+	}
+	const BOOL appendAboutBoxResult = pSysMenu->AppendMenu(MF_STRING, IDM_ABOUTBOX, strAboutMenu);
+	if (appendAboutBoxResult == 0)
+	{
+		outputPrintf(L"Failed to append about box string to menu!\n");
+		return false;
+	}
+	return true;
+}
+
+void checkETWCompatibility( )
+{
+	if (!IsWindowsVistaOrGreater())
+	{
+		AfxMessageBox(L"ETW tracing requires Windows Vista or above.");
+		exit(10);
+	}
+	return;
+}
+
+void CheckDialogButtons(_In_ CWnd* const dlgToCheck, _In_ const bool bCompress,

```

Comment:
This has the same problem as the original tooltips button - it pretends to not be tightly coupled to the class, but it is. Having this as a non-member function means that it is more error prone and harder to update. It should be a member function.